presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -249,35 +254,61 @@ private Optional<PlanNode> coalesceWithNullAggregation(AggregationNode aggregati
         AggregationNode aggregationOverNull = aggregationOverNullInfo.getAggregation();
         Map<VariableReferenceExpression, VariableReferenceExpression> sourceAggregationToOverNullMapping = aggregationOverNullInfo.getVariableMapping();
 
-        // Do a cross join with the aggregation over null
-        JoinNode crossJoin = new JoinNode(
-                outerJoin.getSourceLocation(),
-                idAllocator.getNextId(),
-                JoinNode.Type.INNER,
-                outerJoin,
-                aggregationOverNull,
-                ImmutableList.of(),
-                ImmutableList.<VariableReferenceExpression>builder()
-                        .addAll(outerJoin.getOutputVariables())
-                        .addAll(aggregationOverNull.getOutputVariables())
-                        .build(),
-                Optional.empty(),
-                Optional.empty(),
-                Optional.empty(),
-                Optional.empty(),
-                ImmutableMap.of());
+        FunctionResolution functionResolution = new FunctionResolution(functionAndTypeManager.getFunctionAndTypeResolver());
+        Map<VariableReferenceExpression, RowExpression> literalMap = new HashMap<>();
+
+        if (useDefaultsForCorrelatedAggregations) {
+            for (Map.Entry<VariableReferenceExpression, AggregationNode.Aggregation> aggregation : aggregationNode.getAggregations().entrySet()) {
+                FunctionHandle functionHandle = aggregation.getValue().getFunctionHandle();
+
+                RowExpression defaultLiteral = null;
+                if (functionResolution.isCountFunction(functionHandle)) {
+                    defaultLiteral = constant(Long.valueOf(0), aggregation.getKey().getType());
+                }
+                else if (functionResolution.isMaxFunction(functionHandle) || functionResolution.isMinFunction(functionHandle) ||
```

Comment:
I think for this case we should be able to get the FunctionMetadata and check !isCalledOnNullInput().  The only thing is, I'm not sure these are set properly for count and approx_distinct function (i couldn't see that they were set different from the default of false), so that may need to be fixed.

Reply:
If I understand correctly, you are suggesting that for all functions for which !calledOnNullInput we should be able to replace with constantNull - not just min, max, sum & avg. 
Makes sense to me . I opened another [PR](https://github.com/prestodb/presto/pull/19123) to set isCalledOnNullInput for aggregate functions to avoid cluttering up this PR. Let's get that merged prior to this

Reply:
> I'm not sure these are set properly for count and approx_distinct function 

isCalledOnNullInput resolves to `false` for these functions. See [this doc](https://docs.google.com/spreadsheets/d/1H3Q11HqK8k9AWOUP_jELISttdPUQsL58SWCvWPyKQxY/edit#gid=375697862) and related [PR comment](https://github.com/prestodb/presto/pull/18938#discussion_r1111558460)

Reply:
yeah, i think we can leave to just this list for now until we fix this field for aggregation functions that need it.  Ideally it would be nice to be able to make this more expansive and check isCalledOnNullInput

Reply:
oh just saw your other PR does go through all aggregations to set this correctly.  It's marked as a draft. Is it ready for review now or should I wait?

Reply:
Just having trouble getting a clean test run